go/libraries/doltcore/sqle/dfunctions/verify_constraints.go

```diff
@@ -55,6 +56,9 @@ func (vc *ConstraintsVerifyFunc) Eval(ctx *sql.Context, row sql.Row) (interface{
 }
 
 func DoDoltConstraintsVerify(ctx *sql.Context, args []string) (int, error) {
+	if err := branch_control.CheckAccess(ctx, branch_control.Permissions_Write); err != nil {
```

Comment:
Does this need write?

Reply:
I think so, verifying constraints can modify the system tables for that branch, so it _can_ do a write operation on the branch.